pages/_app.js

```diff
@@ -2,17 +2,142 @@ import GlobalStyle from "../styles";
 import { useImmerLocalStorageState } from "@/hooks/useImmerLocalStorageState";
 import { contactsSampleData } from "@/data/SampleData";
 import Layout from "@/components/Layout/Layout";
+import { uid } from "uid";
+import { useRouter } from "next/router";
 
 export default function App({ Component, pageProps }) {
+  const router = useRouter();
   const [contacts, updateContacts] = useImmerLocalStorageState("contacts", {
     defaultValue: contactsSampleData,
   });
 
+  function handleAddNewContact(event) {
+    event.preventDefault();
+
+    const formData = new FormData(event.target);
+    const newContact = Object.fromEntries(formData);
+
+    const newContactId = uid();
+
+    const date = new Date();
+    const currentUtcDateTime = date.toISOString();
+
```

Comment:
all this form handling should be done in your form component. Try to keep the app.js as clean as possible.

Reply:
Moved almost all of handleAddNewContact to ContactForm.js "handleSubmitForm" function